GETTING_STARTED.md

```diff
@@ -44,7 +44,7 @@ Here are some examples of first PRs from existing contributors:
 The first step is finding a missing optimisation pattern.
 You're welcome to come up with your own, or do one of the following:
 
-- `x * 2ⁱ -> x << i`
+- `x + x * n -> x * (n + 1)` (where `n` is an immediate)
```

Comment:
My current gut feeling is that the previous suggestion was not very good, as it's a valid optimisation (on some targets), but not a great canonicalization (since it actually makes code harder to process by the compiler than the input). I think the changes in the rest of the PR are already evidence of you getting the APIs etc, which was most of the point of the exercise, and that it would be ideal to just merge the change to this line, proposing a better canonicalization pattern.

Reply:
BTW it's better because in the LHS, x is used twice, and in the rhs it's used once, which may unblock some optimisations that check the number of uses of an ssa value.